### PR TITLE
Remove `T.any(nil, ....)` when getting the return type of a block

### PR DESCRIPTION
**Pull request by neilparikh**
### Motivation

 Fixes https://github.com/sorbet/sorbet/issues/1767

Since the proc was wrapped in `T.nilable`, the return type would be set at `T.untyped` instead of `Integer`.

### Test plan

See included automated tests.

### CODE REVIEW

File: core/types/types.cc [hunk 1]
```diff
@@ -614,10 +614,12 @@ TypePtr Types::resultTypeAsSeenFrom(Context ctx, TypePtr what, SymbolRef fromWha
 }
 
 TypePtr Types::getProcReturnType(Context ctx, const TypePtr &procType) {
```
Comment (DarkDimius) [inline core/types/types.cc]: This seems to be a general utility function that has a general name. Having it do a `NilClass` substraction seems surprising given general name. Can we create a separate helper for this?

Comment (neilparikh) [reply core/types/types.cc]: Ya, I found it a little odd to do the subtraction here as well. Are you proposing something like the following (at the caller)?

```
auto tp = ...

blockReturnTp = getProcReturnType(subtractNilHelper(tp))
```

Comment (DarkDimius) [reply core/types/types.cc]: yup, 
+  `subtractNilHelper` is `dropSubtypesOf(ctx, tp, Symbols::NilClass())`

**Comment by connorshea** [comment]
🙏 I'd really love to see this merged.

File: core/types/types.cc (hunk 1 above)
Comment (neilparikh) [reply core/types/types.cc]: Made a `dropNil` helper function.

**Comment by ptarjan** [comment]
@neilparikh sadly this failed linters. Can you run the linters please?

**Pull request merged**